cmds/vpdbootmanager/add.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/systemboot/systemboot/pkg/booter"
+	"github.com/systemboot/systemboot/pkg/vpd"
+)
+
+var dryRun = false
+
+func add(entrytype string, args []string) error {
+	var entry booter.Booter
+	var err error
+	switch entrytype {
+	case "netboot":
+		if len(args) < 2 {
+			return fmt.Errorf("You need to pass method and MAC address")
+		}
+		entry, err = parseNetbootFlags(args[0], args[1], args[2:])
+		if err != nil {
+			return err
+		}
+	case "localboot":
+		if len(args) < 1 {
+			return fmt.Errorf("You need to provide method")
+		}
+		entry, err = parseLocalbootFlags(args[0], args[1:])
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("Unknown entry type")
+	}
+	if dryRun {
```

Comment:
this (global var) is one of the reasons why I'd have preferred to separate the generation of the entry from writing it, but I guess we can live with it

Reply:
Generation and writing **is** separated. This global var is beacuse go `flag` package is not smart enough to handle "subcommand" case. So we need to use two flagsets (one for each subcommand (localboot/netboot), beacuse they have different flags). So, to handle this case and not use any smarter flag-parser that support subcommand we need to store "global" flags in variable.